pandas/tests/indexes/multi/test_indexing.py

```diff
@@ -0,0 +1,221 @@
+        idx.slice_locs(timedelta(seconds=30))
+    # TODO: Try creating a UnicodeDecodeError in exception message
+    with tm.assert_raises_regex(TypeError, '^Level type mismatch'):
+        idx.slice_locs(df.index[1], (16, "a"))
+
+
+def test_slice_locs_not_sorted():
+    index = MultiIndex(levels=[Index(lrange(4)), Index(lrange(4)), Index(
+        lrange(4))], labels=[np.array([0, 0, 1, 2, 2, 2, 3, 3]), np.array(
+            [0, 1, 0, 0, 0, 1, 0, 1]), np.array([1, 0, 1, 1, 0, 0, 1, 0])])
+
+    tm.assert_raises_regex(KeyError, "[Kk]ey length.*greater than "
+                           "MultiIndex lexsort depth",
+                           index.slice_locs, (1, 0, 1), (2, 1, 0))
+
+    # works
+    sorted_index, _ = index.sortlevel(0)
+    # should there be a test case here???
+    sorted_index.slice_locs((1, 0, 1), (2, 1, 0))
+
+
+def test_slice_locs_not_contained():
+    # some searchsorted action
+
+    index = MultiIndex(levels=[[0, 2, 4, 6], [0, 2, 4]],
+                       labels=[[0, 0, 0, 1, 1, 2, 3, 3, 3],
+                               [0, 1, 2, 1, 2, 2, 0, 1, 2]], sortorder=0)
+
+    result = index.slice_locs((1, 0), (5, 2))
+    assert result == (3, 6)
+
+    result = index.slice_locs(1, 5)
+    assert result == (3, 6)
+
+    result = index.slice_locs((2, 2), (5, 2))
+    assert result == (3, 6)
+
+    result = index.slice_locs(2, 5)
+    assert result == (3, 6)
+
+    result = index.slice_locs((1, 0), (6, 3))
+    assert result == (3, 8)
+
+    result = index.slice_locs(-1, 10)
+    assert result == (0, len(index))
+
+
+def test_to_series(idx):
+    # assert that we are creating a copy of the index
+
+    s = idx.to_series()
+    assert s.values is not idx.values
+    assert s.index is not idx
+    assert s.name == idx.name
+
+
+def test_to_series_with_arguments(idx):
+    # GH18699
+
+    # index kwarg
+    s = idx.to_series(index=idx)
+
+    assert s.values is not idx.values
+    assert s.index is idx
+    assert s.name == idx.name
+
+    # name kwarg
+    idx = idx
+    s = idx.to_series(name='__test')
+
+    assert s.values is not idx.values
+    assert s.index is not idx
+    assert s.name != idx.name
+
+
+def test_shift(idx):
+
+    # GH8083 test the base class for shift
+    pytest.raises(NotImplementedError, idx.shift, 1)
+    pytest.raises(NotImplementedError, idx.shift, 1, 2)
+
+
+def test_insert_base(idx):
+
+    result = idx[1:4]
+
+    # test 0th element
+    assert idx[0:4].equals(result.insert(0, idx[0]))
+
+
+def test_delete_base(idx):
+
+    expected = idx[1:]
+    result = idx.delete(0)
+    assert result.equals(expected)
+    assert result.name == expected.name
+
+    expected = idx[:-1]
+    result = idx.delete(-1)
+    assert result.equals(expected)
+    assert result.name == expected.name
+
+    with pytest.raises((IndexError, ValueError)):
+        # either depending on numpy version
+        result = idx.delete(len(idx))
+
+
+def test_fillna(idx):
```

Comment:
can you make a test_missing, move fillna, dropna there

Reply:
Done